Making development easier using docker

jhhammer opened the pull request:
I ran into problems with `npm install` on my machine. So I added functionality and a `README.md` in `docker/README.md` to show for others like me, how they can use docker and easily contribute to this repository.

albanm commented:
I have mixed feelings for this PR.

On one side I am all for making developers lives easier and I often use docker to solve system discrepancies, so I understand what you are doing.

On the other side it feels a little bit overkill with new files requiring future maintenance. This module just depends on a basic nodejs environment, to my knowledge it doesn't require specific native libraries or stuff like that. So this should be accessible on all common platforms.

Was the original problem because of linux syntax in the commands in package.json ? Could it be solved using some small modifications here, maybe using [cross-env](https://www.npmjs.com/package/cross-env) ?

jhhammer commented:
I updated to your last changes and added some improvements to the Dockerfile to use the original unchanged package.json.

The problem is not linux syntax, it is because of version problems. Nuxt 2.12.2 needs npm >=5.0.0. On my development machine I use Ubuntu LTS 18.04 which only has npm v.3.5.2 in the repositories. There is no backport for npm. Because I use docker for almost every project, I simply used the image of Ubuntu LTS 20.04 which contains a newer npm, but it is of course possible to update npm on 18.04 by `sudo npm install -g npm@latest`.

Since I understand your "mixed" feelings, I will close this pull request.
